### PR TITLE
LA-2252,LA2254 Add and remove functionality modified according to our need.

### PR DESCRIPTION
**Pull request by pepcus-Aish**
Add and remove functionality has been added as per the need of our library.

before we could delete each node present and that will eventually result in a structure change of data.
Now we have restricted the delete functionality from each element.

As of now when we hover it will not display the cross icon to delete each node.
hover array node

But we have left the scope to delete the child node of the array of object cases only.
As only in that case when we hover it shows the cross icon.

We have added the plus icon for an array of object case parent node only other than this case, now user cannot add any node so that the structure remains intact of JSON.
plus icon
 
In the **above screenshot** the "**possibleCountryCode**" is an array that shows plus icon to add child node and this functionality is removed from each other node.

### CODE REVIEW

File: src/lib/JSONEditor.js [hunk 1]
```diff
@@ -136,7 +136,7 @@ export default class JSONEditor extends React.Component {
             value={label}
             addElement={this.addElement}
             removeElement={this.removeElement}
-            showRemoveButton={this.props.showRemoveButton}
+          //showRemoveButton={this.props.showRemoveButton}
```
Comment (Nikmahajan) [inline src/lib/JSONEditor.js]: Can we pass value instead!

Comment (pepcus-Aish) [reply src/lib/JSONEditor.js]: @Nikmahajan started to work on passing values as per our discussion.

Comment (pepcus-Aish) [reply src/lib/JSONEditor.js]: @Nikmahajan, Now no code is commented we have added more props to handle our case it is now controllable and values are being passed by props only.

props are added for our cases

we could see showAddButtonObject and ShowRemoveButtonObject is added. to handle lib as per any case.

image

The above code shows no code commented.

File: src/lib/JSONEditor.js [hunk 2]
```diff
@@ -136,7 +138,7 @@ export default class JSONEditor extends React.Component {
             value={label}
             addElement={this.addElement}
             removeElement={this.removeElement}
-            showRemoveButton={this.props.showRemoveButton}
+          showRemoveButton={this.props.showRemoveButton}
```
Comment (Nikmahajan) [inline src/lib/JSONEditor.js]: Please Check indentations!

**Review by Nikmahajan — APPROVED** [review]

File: src/lib/JSONEditor.js (hunk 2 above)
Comment (pepcus-Aish) [reply src/lib/JSONEditor.js]: Thanks Nikhil, The indentations are now taken care of.

**Pull request merged**